app/javascript/dashboard/routes/dashboard/settings/macros/MacroForm.vue

```diff
@@ -109,19 +99,29 @@ export default {
       });
     },
     deleteNode(index) {
+      // delete that index specifically
+      // so that the next item does not get marked invalid
+      delete this.errors[`action_${index}`];
       this.macro.actions.splice(index, 1);
     },
     submit() {
-      this.$v.$touch();
-      if (this.$v.$invalid) return;
+      this.errors = validateActions(this.macro.actions);
+      if (Object.keys(this.errors).length !== 0) return;
+
+      this.v$.$touch();
+      if (this.v$.$invalid) return;
+
       this.$emit('submit', this.macro);
     },
     resetNode(index) {
-      this.$v.macro.actions.$each[index].$reset();
+      // delete that index specifically
+      // so that the next item does not get marked invalid
+      delete this.errors[`action_${index}`];
```

Comment:
const { [`action_${index}`]: _, ...restOfTheErrors} = this.errors
this.errors = restOfTheErrors.

Reply:
Done!